src/ripple/net/impl/RPCCall.cpp

```diff
@@ -495,6 +495,48 @@ class RPCParser
         return rpcError (rpcINVALID_PARAMS);
     }
 
+    bool isValidJson2(Json::Value const& jv)
+    {
+        if (jv.isObject())
+        {
+            if (jv.isMember("jsonrpc") && jv["jsonrpc"] == "2.0" &&
+                jv.isMember("ripplerpc") && jv["ripplerpc"] == "2.0" &&
+                jv.isMember("id") && jv.isMember("method"))
+                return true;
+        }
+        return false;
+    }
+
+    Json::Value parseJson2(Json::Value const& jvParams)
+    {
+        Json::Reader reader;
+        Json::Value jv;
+        reader.parse(jvParams[0u].asString(), jv);
+        if (isValidJson2(jv))
+        {
+            Json::Value jv1{Json::objectValue};
+            jv1["jsonrpc"] = jv["jsonrpc"];
+            jv1["ripplerpc"] = jv["ripplerpc"];
+            jv1["id"] = jv["id"];
+            jv1["method"] = jv["method"];
+            if (jv.isMember("params"))
+            {
+                auto const& params = jv["params"][0u];
```

Comment:
> I don't know the rationale for this API detail.

I think the JSON-RPC 1.0 spec required it in Section 1.1. http://json-rpc.org/wiki/specification
> `params` - An Array of objects to pass as arguments to the method. 

Whereas JSON-RPC 2.0 spec allows Array or Object. http://www.jsonrpc.org/specification#parameter_structures
> If present, parameters for the rpc call MUST be provided as a Structured value. Either by-position through an Array or by-name through an Object.

Before it was _required_ to be an Array. Now that it's _allowed_ to be an Object, I think it makes much more sense to use an Object.